merkle-tree/bounded-vec/src/lib.rs

```diff
@@ -341,6 +358,108 @@ where
         }
         Ok(())
     }
+
+    pub fn deserialize(
+        account_data: &mut [u8],
+        start_offset: &mut usize,
+    ) -> Result<ManuallyDrop<BoundedVec<T>>, BoundedVecError> {
+        unsafe {
+            if account_data.len().saturating_sub(*start_offset)
+                < std::mem::size_of::<BoundedVecMetadata>()
+            {
+                return Err(BoundedVecError::InsufficientMemoryAllocated(
+                    account_data.len().saturating_sub(*start_offset),
+                    std::mem::size_of::<BoundedVecMetadata>(),
+                ));
+            }
+            let metadata: *mut BoundedVecMetadata = read_ptr_at(account_data, start_offset);
+            if ((*metadata).capacity() * std::mem::size_of::<T>()) % 8 != 0 {
+                return Err(BoundedVecError::UnalignedMemory(
+                    (*metadata).capacity() * std::mem::size_of::<T>(),
+                ));
+            }
+            let full_vector_size = std::mem::size_of::<BoundedVecMetadata>()
+                + ((*metadata).capacity() * std::mem::size_of::<T>());
+            if account_data.len().saturating_sub(*start_offset) < full_vector_size {
+                return Err(BoundedVecError::InsufficientMemoryAllocated(
+                    account_data.len().saturating_sub(*start_offset),
+                    full_vector_size,
+                ));
+            }
+            Ok(ManuallyDrop::new(BoundedVec::from_raw_parts(
+                metadata,
+                read_array_like_ptr_at(account_data, start_offset, (*metadata).capacity()),
+            )))
+        }
+    }
+
+    pub fn deserialize_multiple(
+        num: usize,
+        account_data: &mut [u8],
+        start_offset: &mut usize,
+    ) -> Result<Vec<ManuallyDrop<BoundedVec<T>>>, BoundedVecError> {
+        let mut value_vecs = Vec::with_capacity(num);
+        for _ in 0..num {
+            let vec = Self::deserialize(account_data, start_offset)?;
+            value_vecs.push(vec);
+        }
+        Ok(value_vecs)
+    }
+
+    pub fn init(
+        capacity: usize,
+        account_data: &mut [u8],
+        start_offset: &mut usize,
+        with_len: bool,
+    ) -> Result<ManuallyDrop<BoundedVec<T>>, BoundedVecError> {
+        let vector_size = capacity * std::mem::size_of::<T>();
+        if vector_size % 8 != 0 {
```

Comment:
I would change it to

```suggestion
        if vector_size % mem::align_of::<T>() != 0 {
```